api/nodes/serializers.py

```diff
@@ -266,6 +269,111 @@ def update(self, node, validated_data):
         return node
 
 
+class NodeAddonSettingsSerializer(JSONAPISerializer):
+    """
+    Overrides AddonSettingsSerializer to return node-specific fields
+    """
+    class Meta:
+        type_ = 'node_addons'
+
+    id = ser.CharField(source='_id', read_only=True)
+    enabled = ser.BooleanField(required=True)
+    auth_id = ser.CharField(source='object.external_account._id', allow_null=True)
+    folder_id = ser.CharField(source='object.folder_id', allow_null=True)
+    folder_path = ser.CharField(source='object.folder_path', required=False, allow_null=True)
+    has_auth = ser.BooleanField(source='object.has_auth', read_only=True)
+    complete = ser.BooleanField(source='object.complete', read_only=True)
+    configured = ser.BooleanField(source='object.configured', read_only=True)
+    node = ser.CharField(source='object.owner._id', read_only=True)
```

Comment:
This should likely be a relationship link to a node rather than just the node id unless that's super inconvenient for the major use case / every use case.